set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/ElevationUserSetRest.java

```diff
@@ -0,0 +1,156 @@
+    verifyReadAccess(request);
+    return generateElevation();
+    }
+
+    /**
+     * generate elevation file
+     * @return
+     * @throws HttpException
+     */
+    private ResponseEntity<String> generateElevation() throws HttpException {
+    try {
+        List<PersistentUserSet> usersets = getUserSetService().getEntitySetBestBetsItems();
+        Elevation elevation = buildElevation(usersets);
+        if (elevation != null) {
+            UserSetXMLSerializer xmlSerializer = new UserSetXMLSerializer();
+            String xml = xmlSerializer.serialize(elevation);
+            writeElevation(getConfiguration().getElevationFileLocation(), xml);
+            // returning the elevation response
+            //TODO - remove the body, once we know how elevation file will be used
+            return new ResponseEntity<>(xml, HttpStatus.OK);
+        } else {
+            throw new UserSetNotFoundException(UserSetI18nConstants.ELEVATION_NOT_GENERATED,
+                    UserSetI18nConstants.ELEVATION_NOT_GENERATED,
+                    null);
+        }
+    } catch (IOException e) {
+        throw new InternalServerException(e);
+    }
+    }
+
+    /**
+     * Build elevation object
+     * @param userSets
+     * @return
+     */
+    private Elevation buildElevation(List<PersistentUserSet> userSets) throws UserSetNotFoundException {
+    if (userSets.isEmpty()) {
+        throw new UserSetNotFoundException(UserSetI18nConstants.ENTITY_USER_SET_NOT_FOUND,
+                    UserSetI18nConstants.ENTITY_USER_SET_NOT_FOUND,
+                    null);
+        }
+    Elevation elevation = new Elevation();
+    List<Query> queries = new ArrayList<>();
+    for (UserSet userset : userSets) {
+        String text = generateQueryText(userset.getSubject());
+        List<Doc> docList = new ArrayList<>();
+        if (userset.getItems() != null) {
+            for(String item : userset.getItems()) {
+            docList.add(new Doc(item));
+            }
+        }
+        if (!text.isEmpty() && !docList.isEmpty()) {
+            queries.add(new Query(text, docList));
+        }
+    }
+    if (!queries.isEmpty()) {
+        elevation.setQuery(queries);
+        return elevation;
+    }
+    return null;
+    }
+
+    /**
+     * get the entity reference from the subject list
+     * @param subject
+     * @return
+     */
+    private String generateQueryText(List<String> subject) {
+      StringBuilder text = new StringBuilder();
+      for(String s : subject) {
+          if (s.contains(WebUserSetFields.AGENT)) {
+              text.append(WebUserSetFields.ELEVATION_AGENT_QUERY);
+          }
+          else if (s.contains(WebUserSetFields.CONCEPT)) {
+              text.append(WebUserSetFields.ELEVATION_CONCEPT_QUERY);
+          }
+          else if(s.contains(WebUserSetFields.TIMESPAN)) {
+              text.append(WebUserSetFields.ELEVATION_TIMESPAN_QUERY);
+          } else {
+              break;
```

Comment:
this break is not needed when the for loop is removed

Reply:
yes